Local WebKit builds don't use locally-built WebKit process extensions

pvollan opened the pull request:
#### 52f67b99e6e8949eb858922e818472f4a3089625
<pre>
Local WebKit builds don&apos;t use locally-built WebKit process extensions
<a href="https://bugs.webkit.org/show_bug.cgi?id=273550">https://bugs.webkit.org/show_bug.cgi?id=273550</a>
<a href="https://rdar.apple.com/125726458">rdar://125726458</a>

Reviewed by Elliott Williams.

When locally built WebKit extensions are installed in the app bundle, these extensions
will be launched instead of the system extensions when launching said app. This patch
adds a build step that copies extensions to WebKitTestRunner app bundle.

* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::isInWebKitChildProcess):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::hasExtensionInAppBundle):
(WebKit::launchWithExtensionKit):
* Tools/Scripts/webkitpy/port/device_port.py:
(DevicePort._install_extensions):
(DevicePort._install):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/279946@main">https://commits.webkit.org/279946@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a058e344cb91706c8d64c6fee50b6c969f05891c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34439 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7575 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58260 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5713 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57282 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41989 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5745 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44529 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3887 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57077 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32525 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47636 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25655 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29311 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4980 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3854 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/51169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5201 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59851 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30249 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5357 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51951 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31379 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47712 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51390 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32400 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8142 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31170 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
```diff
@@ -37,13 +37,19 @@ STRIP_STYLE = debugging;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=macosx*] = ios/* AppDelegate.m *.appex;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=appletv*] = ios/Launch.storyboard *.appex;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=watch*] = ios/Launch.storyboard *.appex;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=xr*] = *.appex;
```
As written, I think the `config` and `sdk` variations will override each other. You could add an `$(inherited)` to both lines so that whichever one is evaluated first does not override the other.

emw-apple commented on Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
```diff
@@ -37,13 +37,19 @@ STRIP_STYLE = debugging;
+
+APPLY_RULES_IN_COPY_FILES = YES
+
+WK_APP_EXTENSION_BUILD_DIR = $(BUILT_PRODUCTS_DIR);
```
Seems like this setting is not needed anymore.

emw-apple commented on Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
```diff
@@ -37,13 +37,19 @@ STRIP_STYLE = debugging;
 CODE_SIGN_ENTITLEMENTS[sdk=watch*] = Configurations/WebKitTestRunnerApp-watchOS.entitlements;
 CODE_SIGN_ENTITLEMENTS[sdk=xr*] = Configurations/WebKitTestRunnerApp-iOS.entitlements;
+
+APPLY_RULES_IN_COPY_FILES = YES
```
nit: we always put a semicolon at the end of xcconfig lines

emw-apple reviewed (approved):
r+, though this needs to wait until internal changes are ready

pvollan committed:
Local WebKit builds don't use locally-built WebKit process extensions
https://bugs.webkit.org/show_bug.cgi?id=273550
rdar://125726458

Reviewed by Elliott Williams.

When locally built WebKit extensions are installed in the app bundle, these extensions
will be launched instead of the system extensions when launching said app. This patch
adds a build step that copies extensions to WebKitTestRunner app bundle.

* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::isInWebKitChildProcess):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::hasExtensionInAppBundle):
(WebKit::launchWithExtensionKit):
* Tools/Scripts/webkitpy/port/device_port.py:
(DevicePort._install_extensions):
(DevicePort._install):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/279946@main
